File: lib/services/repository/mobile_connecting_services/user_repository.dart
import 'package:dio/dio.dart';
import '../../../utils/dio_client.dart';

class UserRepository {
  final DioClient dioClient;

  UserRepository({required this.dioClient});

  Future<Map<String, dynamic>> login(String email, String password) async {
    final response = await dioClient.post('/login', {
      'email': email,
      'password': password,
    });
    return response.data;
  }

  Future<Map<String, dynamic>> register(String email, String password) async {
    final response = await dioClient.post('/register', {
      'email': email,
      'password': password,
    });
    return response.data;
  }
}

File: lib/services/repository/python_backend/user_repository.py
import requests

class UserRepository:
    def __init__(self, base_url):
        self.base_url = base_url

    def login(self, email, password):
        response = requests.post(f"{self.base_url}/login", json={"email": email, "password": password})
        if response.status_code == 200:
            return response.json()
        else:
            return None

    def register(self, email, password):
        response = requests.post(f"{self.base_url}/register", json={"email": email, "password": password})
        if response.status_code == 200:
            return response.json()
        else:
            return None